example/src/main/java/io/netty/example/http2/tiles/http2/ImageCache.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2015 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.netty.example.http2.tiles.http2;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.commons.io.IOUtils;
+
+/**
+ * Caches the images to avoid reading them every time from the disk.
+ */
+public final class ImageCache {
+
+    private static ImageCache INSTANCE;
+
+    private final Map<String, byte[]> imageBank = new HashMap<String, byte[]>(200);
+
+    private ImageCache() {
+    }
+
+    public static synchronized ImageCache instance() {
+        if (INSTANCE == null) {
+            INSTANCE = new ImageCache();
+            INSTANCE.initImageBank();
+        }
+        return INSTANCE;
+    }
+
+    public static String name(int x, int y) {
+        return "tile-" + y + "-" + x + ".jpeg";
+    }
+
+    public byte[] image(int x, int y) {
+        return imageBank.get(name(x, y));
+    }
+
+    private void initImageBank() {
+        for (int y = 0; y < 10; y++) {
+            for (int x = 0; x < 20; x++) {
+                try {
+                    String name = name(x, y);
+                    byte[] fileBytes = IOUtils.toByteArray(getClass().getResourceAsStream(name));
```

Comment:
Can we use netty `ByteBuf` and/or `ByteBufUtils` to do this instead of pulling in commons.io?